### PR TITLE
Replace Clients with Writers, StatusWriters and APIReaders

### PR DESCRIPTION
**Pull request by hatfieldbrian**
Non-API server reads have caused problems more than once:
- #239 
- #440 
- #477 
- #496 

This patch replaces Clients with Writers, StatusWriters, and APIReaders making it less likely to do non-API server reads.

### CODE REVIEW

File: controllers/drcluster_controller.go [hunk 1]
```diff
@@ -40,7 +40,8 @@ import (
 
 // DRClusterReconciler reconciles a DRCluster object
 type DRClusterReconciler struct {
-	client.Client
+	client.Writer
+	client.StatusWriter
```
Comment (BenamarMk) [inline controllers/drcluster_controller.go:44]: The create,update,delete, etc are already a client.Writer members.  In other words, there is no need to explicitly call the Update() function with client.Writer object.  client.Client.Update() will do the same the same.

**Review by BenamarMk — COMMENTED** [review]
I don't think I agree with this change. Using the APIReader for all calls is dangerous.  The APIReader should be used selectively.  We use it to read the resource being reconciled.  That's necessary because from the time we update the status of the object to the time the resource is reconciled again could be called within a submillisecond.

**Comment by ShyamsundarR** [comment]
> I don't think I agree with this change. Using the APIReader for all calls is dangerous. The APIReader should be used selectively. We use it to read the resource being reconciled. That's necessary because from the time we update the status of the object to the time the resource is reconciled again could be called within a submillisecond.

Changing to APIReader at most/all places is going to start throttling our requests by the API server. The client cache should be leveraged instead. @hatfieldbrian is there a driving reason for the change?

**Comment by ShyamsundarR** [comment]
Closing as stale, part of general cleanup. Please reopen and update if still required.

**Pull request closed without merging**